service/internal/checksum/middleware_compute_input_checksum_test.go

```diff
@@ -93,33 +93,50 @@ func TestComputeInputPayloadChecksum(t *testing.T) {
 					"CRC32": "AAAAAA==",
 				},
 			},
-			"no algorithm": {
+			"http no algorithm checksum header preset": {
 				buildInput: middleware.BuildInput{
 					Request: func() *smithyhttp.Request {
 						r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
-						r.URL, _ = url.Parse("https://example.aws")
-						r = requestMust(r.SetStream(strings.NewReader("hello world")))
+						r.URL, _ = url.Parse("http://example.aws")
 						r.ContentLength = 11
+						r.Header.Set(AlgorithmHTTPHeader(AlgorithmCRC32), "AAAAAA==")
+						r = requestMust(r.SetStream(bytes.NewReader([]byte("hello world"))))
 						return r
 					}(),
 				},
-				expectHeader:        http.Header{},
+				expectHeader: http.Header{
+					"X-Amz-Checksum-Crc32": []string{"AAAAAA=="},
+				},
 				expectContentLength: 11,
 				expectPayload:       []byte("hello world"),
 			},
-			"nil stream no algorithm require checksum": {
-				optionsFn: func(o *computeInputPayloadChecksum) {
-					o.RequireChecksum = true
-				},
+			"http no algorithm set": {
 				buildInput: middleware.BuildInput{
 					Request: func() *smithyhttp.Request {
 						r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
 						r.URL, _ = url.Parse("http://example.aws")
+						r = requestMust(r.SetStream(strings.NewReader("hello world")))
+						r.ContentLength = 11
 						return r
 					}(),
 				},
-				expectContentLength: -1,
+				expectContentLength: 11,
 				expectHeader:        http.Header{},
+				expectPayload:       []byte("hello world"),
+			},
+			"https no algorithm set": {
```

Comment:
Is there any different treatment between http and https? Unless I'm missing something the tests seem the same

Reply:
they are tested to ensure checksum is calculated and set in header/trailer correctly with different schema cfg